components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -214,20 +217,18 @@ func (sb *sensorBase) SetPower(
 	ctx context.Context, linear, angular r3.Vector, extra map[string]interface{},
 ) error {
 	sb.opMgr.CancelRunning(ctx)
+	if sb.loop != nil {
+		sb.loop.Pause()
+	}
 	return sb.controlledBase.SetPower(ctx, linear, angular, extra)
 }
 
 func (sb *sensorBase) Stop(ctx context.Context, extra map[string]interface{}) error {
 	sb.opMgr.CancelRunning(ctx)
-	sb.stopLoop()
-	return sb.controlledBase.Stop(ctx, extra)
-}
-
-func (sb *sensorBase) stopLoop() {
```

Comment:
okay paused the loop instead of stopping, so removed this function as it no longer felt super necessary